app/controllers/api/functions.php

```diff
@@ -1139,73 +1483,96 @@
         ]);
 
         /** Execute function */
         $executor = new Executor(App::getEnv('_APP_EXECUTOR_HOST'));
         try {
+            $command = 'node src/server.js'; // TODO: Custom for each runtime
             $executionResponse = $executor->createExecution(
                 projectId: $project->getId(),
                 deploymentId: $deployment->getId(),
-                payload: $data,
+                version: $function->getAttribute('version'),
+                body: \strlen($body) > 0 ? $body : null,
                 variables: $vars,
                 timeout: $function->getAttribute('timeout', 0),
                 image: $runtime['image'],
-                source: $build->getAttribute('outputPath', ''),
+                source: $build->getAttribute('path', ''),
                 entrypoint: $deployment->getAttribute('entrypoint', ''),
+                path: $path,
+                method: $method,
+                headers: $headers,
+                runtimeEntrypoint: 'cp /tmp/code.tar.gz /mnt/code/code.tar.gz && nohup helpers/start.sh "' . $command . '"'
             );
 
             /** Update execution status */
-            $execution->setAttribute('status', $executionResponse['status']);
+            $status = $executionResponse['statusCode'] >= 400 ? 'failed' : 'completed';
+            $execution->setAttribute('status', $status);
             $execution->setAttribute('statusCode', $executionResponse['statusCode']);
-            $execution->setAttribute('response', $executionResponse['response']);
-            $execution->setAttribute('stdout', $executionResponse['stdout']);
-            $execution->setAttribute('stderr', $executionResponse['stderr']);
+            $execution->setAttribute('logs', $executionResponse['logs']);
+            $execution->setAttribute('errors', $executionResponse['errors']);
             $execution->setAttribute('duration', $executionResponse['duration']);
         } catch (\Throwable $th) {
-            $interval = (new \DateTime())->diff(new \DateTime($execution->getCreatedAt()));
+            $durationEnd = \microtime(true);
+
             $execution
-                ->setAttribute('duration', (float)$interval->format('%s.%f'))
+                ->setAttribute('duration', $durationEnd - $durationStart)
                 ->setAttribute('status', 'failed')
-                ->setAttribute('statusCode', $th->getCode())
-                ->setAttribute('stderr', $th->getMessage());
+                ->setAttribute('statusCode', 500)
+                ->setAttribute('errors', $th->getMessage() . '\nError Code: ' . $th->getCode());
             Console::error($th->getMessage());
         }
 
-        Authorization::skip(fn () => $dbForProject->updateDocument('executions', $executionId, $execution));
+        if ($function->getAttribute('logging')) {
+            Authorization::skip(fn () => $dbForProject->updateDocument('executions', $executionId, $execution));
+        }
```

Comment:
Can we write to the execution document only once? We do both create an update in the same route.

Reply:
Updated to only do create.